app/controllers/courses_controller.rb

```diff
@@ -229,6 +229,109 @@ def users
             end
   end
 
+  action_auth_level :add_users_from_emails, :instructor
+  def add_users_from_emails
+    params.require([:user_emails, :role])
```

Comment:
I don't think this line is necessary, since you're not using the whole `params` object verbatim